### DISCUSSION TITLE
emmyLua tresitter support?

### DISCUSSION THREAD

**Discussion by rmagatti** [numToStr/lemmy-help#64] [Ideas]
Hey, I noticed this when opening github's `codespaces`.

<img width="1064" alt="Screenshot 2022-11-17 at 1 38 13 AM" src="https://user-images.githubusercontent.com/2881382/202397375-78df6b4c-79de-48fb-8880-61621f76ad86.png">

i.e there is proper highlighting for emmyLua docs.

I didn't know where else to ask this so I thought I'd come here, do you know of an existing emmyLua treesitter grammar around? And most importantly (for me) do you happen to know of a way to use it with Neovim?

If there isn't a treesitter grammar for it, maybe that's something we could collaborate on, I have added a feature to the typescript treesitter parser (literally one) and it was kind of interesting to work on, so adding a whole new grammar should be a potentially even more interesting project if it doesn't already exist.

Anywho, if you read this, thanks for your time! If you don't, still thanks for the work on lemmy-help.

Cheers!

**Comment by numToStr** [comment]
There is only one comment parser https://github.com/stsewd/tree-sitter-comment that I know of which can only highlight the initial tag i.e., `@class`, and I don't think there is any emmylua parser around. I'll be happy to collaborate on but like you, I also know nothing about creating treesitter parser, but I like challenges :)

Also, just to throw info around, technically both emmylua and lua-language-server (v3 ownward) annotations are [not compatible](https://github.com/sumneko/lua-language-server/issues/980) with each other.

 - Emmylua - https://emmylua.github.io/annotation.html
 - LLS's annotations - https://github.com/sumneko/lua-language-server/wiki/Annotations

**Reply by rmagatti** [reply]
Right. I suppose an LLS annotation parser would be what I'd want to go for then

**Comment by rmagatti** [comment]
On second thought, maybe any parsing for lua comments like that should just go in https://github.com/tjdevries/tree-sitter-lua ? Might be a better play than just making a whole new parser 🤔

**Reply by numToStr** [reply]
Ahhh, yeah you are right! You could contribute to that parser, but I am guessing that TJ's parser is not updated for quite some time and might be outdated. Another one would be https://github.com/MunifTanjim/tree-sitter-lua which is now shipped with neovim core. IMO you should start from scratch and play with it, and when you are confident enough with the parser then try to upstream it.
